src/act.other.cpp

```diff
@@ -673,9 +673,11 @@ ACMD(do_shapechange) {
         if (creatures[index].skills[i])
             SET_SKILL(mob, creatures[index].skills[i], roll_skill(mob, creatures[index].skills[i]));
 
+    /* Add up to 10% bonus max hp/max mv based on a player's Charisma. */
+    GET_MAX_HIT(mob) = random_number(creatures[index].minhp, creatures[index].maxhp) * (1 + (GET_CHA(ch) / 100));
```

Comment:
Do you want to do it this way.  Maybe move it outside the random number so they always get the bonus and it's not as much up to the RNG gods.

Reply:
I thought this is outside the rng.  I thought the random_number function was just a roll between min and max limits, then whatever that number is is multiplied by 1.x.

I can reformat it so it's more clear?